### PR TITLE
Add marker popup

### PR DESCRIPTION
**Pull request by KamilMa**

### CODE REVIEW

File: client/src/components/Map/Map.js [hunk 1]
```diff
@@ -58,8 +59,12 @@ export default class Map extends Component {
...
                 key={place.id}
                 offsetTop={-40}
-                offsetLeft={-15}>
-                <FontAwesomeIcon style={{ color: colors.sulu }} icon="map-marker-alt" size="4x"></FontAwesomeIcon>
+                offsetLeft={-15}
+              >
+                <FontAwesomeIcon style={{ color: colors.sulu }} icon="map-marker-alt" size="4x"
+                  data-div_id={place.id} onClick={this.props.startPopup}></FontAwesomeIcon>
+
+                {this.props.showPopup && place.id == this.props.clickedId ? <Popup name={place.name} address={place.address} /> : null}
```
Comment (tkleszczewski) [inline client/src/components/Map/Map.js]: turn not exact equality check (==) to strict equality check (===)

File: client/src/App.js [hunk 2]
```diff
@@ -137,6 +137,8 @@ class App extends Component {
       ],
       events: [],
       users: [],
+      showPopup: false,
+      clickedId: null
```
[the remark this answers is not in the document]
Comment (tkleszczewski) [reply client/src/App.js]: Or even more accurately activePopupId and showPopup -> popupActive but showPopup is still OK :)

[the remark this answers is not in the document]
Comment (KamilMa) [reply client/src/App.js]: Fixed

File: client/src/App.js [hunk 3]
```diff
@@ -167,60 +169,98 @@ class App extends Component {
...
-        {
-          path: '/',
-          component: PlaceList,
-          exact: true,
-          handlers: {},
-          cProps: {
-            places: this.state.places,
-          },
+  startPopup = (e) => {
+    // click this same marker
```
[the remark this answers is not in the document]
Comment (KamilMa) [reply client/src/App.js]: Fixed

File: client/src/App.js [hunk 4]
```diff
@@ -167,60 +169,98 @@ class App extends Component {
...
+    if (this.state.clickedId === e.currentTarget.dataset.div_id) {
+      // add/remove class "on-top" (z-index:999)
```
[the remark this answers is not in the document]
Comment (KamilMa) [reply client/src/App.js]: Fixed

File: client/src/App.js [hunk 5]
```diff
@@ -167,60 +169,98 @@ class App extends Component {
...
+      e.currentTarget.parentNode.classList.toggle('on-top')
+      this.setState({
+        showPopup: !this.state.showPopup
+      })
+    } else {
+      // click new marker
```
[the remark this answers is not in the document]
Comment (KamilMa) [reply client/src/App.js]: Fixed

File: client/src/App.js [hunk 6]
```diff
@@ -167,60 +169,98 @@ class App extends Component {
...
+      if (this.state.clickedId !== null) {
+        // remove class "on-top" from previous marker
```
[the remark this answers is not in the document]
Comment (KamilMa) [reply client/src/App.js]: Fixed

File: client/src/assets/styles/colors.js [hunk 7]
```diff
@@ -15,7 +15,7 @@ const colors = {
   primary: '#2052A2',
   success: '#3DB22F',
   danger: '#A21D19',
-  sulu: '#B2ED61'
+  sulu: '#B2ED61',
```
[the remark this answers is not in the document]
Comment (KamilMa) [reply client/src/assets/styles/colors.js]: Fixed

File: client/src/components/Map/Map.js [hunk 8]
```diff
@@ -44,7 +45,7 @@ export default class Map extends Component {
           <FontAwesomeIcon icon="list" />
         </Link>
       </CommonMobileLink>
-      <ReactMapGl
+      <ReactMapGl onClick={this.props.closePopup}
```
[the remark this answers is not in the document]
Comment (KamilMa) [reply client/src/components/Map/Map.js]: Fixed

File: client/src/components/Map/Map.js [hunk 9]
```diff
@@ -58,8 +59,12 @@ export default class Map extends Component {
                 latitude={place.latitude}
                 key={place.id}
                 offsetTop={-40}
-                offsetLeft={-15}>
-                <FontAwesomeIcon style={{ color: colors.sulu }} icon="map-marker-alt" size="4x"></FontAwesomeIcon>
+                offsetLeft={-15}
+              >
+                <FontAwesomeIcon style={{ color: colors.sulu }} icon="map-marker-alt" size="4x"
```
[the remark this answers is not in the document]
Comment (KamilMa) [reply client/src/components/Map/Map.js]: Fixed

File: client/src/components/Map/Map.js (hunk 1 above)
Comment (KamilMa) [reply client/src/components/Map/Map.js]: Fixed

File: client/src/components/Map/Popup.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+`;
+
+const ListItem = styled.li`
+  &:first-child {
+    margin-bottom: 7px;
+  }
+`;
+
+const Popup = (props) => {
+  const { name, address: { street, number } } = props
```
[the remark this answers is not in the document]
Comment (KamilMa) [reply client/src/components/Map/Popup.js]: :)

**Pull request merged**